CMakeLists.txt

```diff
@@ -1,6 +1,34 @@
 
 cmake_minimum_required(VERSION 3.8)
 
+# Includes in this project are meant for dbt, so they are relative to dbt's DEPS folder,
+# where all modules, including the main project, would have their own folder.
+# So every include to files in this project is prefixed by the name of the project folder.
+# For example, to include io/io.hh we use #include "libsupxx/io/io.hh"
+# 
+# To make CMake understand this structure, we recreate the DEPS folder inside CMake's 
+# ${CMAKE_BINARY_DIR} folder. This is NOT in the same place as dbt's DEPS or BUILD folder,
+# so CMake and dbt are kept separate, and can be used at the same time.
+
+# Make the CMAKE_DEPS folder (full path ${DBT_INCLUDE_ROOT}) where we recreate dbt's DEPS
+set(DBT_INCLUDE_ROOT ${CMAKE_BINARY_DIR}/CMAKE_DEPS)
```

Comment:
I guess `CMAKE_DEPS` is inside the `build` folder for cmake? Otherwise it should be added to gitignore.

Reply:
Yes, it is